scarlet/initialization.py

```diff
@@ -157,27 +189,71 @@ def trim_morphology(center_index, morph, bg_thresh):
                 bbox.stop[1] - center_index[-1],
             )
         )
-        while boxsize < size:
-            boxsize += 16  # keep box sizes quite small
+    else:
+        size = 0
 
-    # define bbox and trim to bbox
+    # define new box and cut morphology accordingly
+    boxsize = get_minimal_boxsize(size)
     bottom = center_index[0] - boxsize // 2
     top = center_index[0] + boxsize // 2
     left = center_index[1] - boxsize // 2
     right = center_index[1] + boxsize // 2
     bbox = Box.from_bounds((bottom, top), (left, right))
     morph = bbox.extract_from(morph)
-    bbox = Box.from_bounds((bottom, top), (left, right))
     return morph, bbox
 
 
+def init_compact_source(
+    sky_coord, frame, observations,
+):
+
+    # get PSF-corrected center pixel spectrum
+    spectrum = get_pixel_spectrum(sky_coord, observations, correct_psf=True)
```

Comment:
why not use `get_psf_spectrum` here? Isn't this exactly the case that function is meant to handle?

Reply:
No, these two are different. The `get_psf_spectrum` measures the spectrum with a (target) PSF-shaped window function. `get_pixel_spectrum` only uses a single pixel, but then has to correct for PSF shape variations.

Reply:
I understand that, but I thought that for a point source `get_psf_spectrum` will give the exact spectral model. Is your argument that since a `CompactSource` is not necessarily a point source that using `get_psf_spectrum` would be introducing a larger error? Or am I still not understanding something?

Reply:
It's still different. We _never_ use `get_psf_spectrum`. Even `PointSource` uses `get_pixel_spectrum` because it is more reliable in blends. If the aperture is (observation)PSF-sized, there can be a whole lot of light from neighboring coming in.

The point here is that for _isolated_ point sources, these two functions (with PSF correction) yield the same. In the isolated case `get_psf_spectrum` is superior because it uses a larger aperture is is thus more robust to noise.